components/script/dom/nodelist.rs

```diff
@@ -277,3 +284,18 @@ impl ChildrenList {
         self.last_index.set(0u32);
     }
 }
+
+pub struct NodeListIterator<'a> {
+    nodes: &'a NodeList,
+    offset: u32,
+}
+
+impl<'a> Iterator for NodeListIterator<'a> {
+    type Item = Root<Node>;
+
+    fn next(&mut self) -> Option<Root<Node>> {
+        let result = self.nodes.Item(self.offset);
+        self.offset = self.offset + 1;
+        result
+    }
+}
```

Comment:
Why do you need this iterator? We shouldn't use NodeList outside from other DOM stuff IMO, it's quite inefficient.

Also, this iterator continues to increment offset for nothing if `self.nodes.Item(self.offset)` returned `None`, and it doesn't provide a size hint so it's probably used inefficiently.